access_error.go

```diff
@@ -34,10 +34,19 @@ func (f *Fosite) WriteAccessError(rw http.ResponseWriter, _ AccessRequester, err
 func (f *Fosite) writeJsonError(rw http.ResponseWriter, err error) {
 	rw.Header().Set("Content-Type", "application/json;charset=UTF-8")
 
-	rfcerr := ErrorToRFC6749Error(err)
+	// We dereference to make a copy and do not change passed err when it is already RFC6749Error.
+	rfcerr := *ErrorToRFC6749Error(err)
 	if !f.SendDebugMessagesToClients {
 		rfcerr.Debug = ""
 	}
+	// We expose both error hint and debug strings through standard error description, too
+	// (they are non-standard fields and some clients do not show them).
+	if rfcerr.Hint != "" {
+		rfcerr.Description += ": " + rfcerr.Hint
```

Comment:
It might be better to add `GetDescription` to rfcerr?

Reply:
Depends, `GetDescription` will not be json marshalled!

Reply:
Yea, but you would use it to construct the JSON.

Reply:
I don't follow? This is JSON Encoded a few lines later: https://github.com/ory/fosite/pull/460/files#diff-49ae52d32b52cf990340c8dbf949704fR53